core/arch/arch.h

```diff
@@ -1336,9 +1336,20 @@ new_thread_setup(priv_mcontext_t *mc);
 #    ifdef MACOS
 void
 new_bsdthread_setup(priv_mcontext_t *mc);
+// Enable writing to MAP_JIT pages.
+#        define pthread_jit_write() pthread_jit_write_protect_np(false)
+// Enable executing MAP_JIT pages.
```

Comment:
Explain that this is per-thread as the general intuition is that it would be process-wide and DR would have to serialize all the threads.